File: backend/src/main/java/com/verifime/dto/Invoice.java
package com.verifime.dto;

import java.time.LocalDate;
import java.util.List;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;

public record Invoice(
        @NotBlank(message = "Invoice currency is required") String currency,
        @NotNull(message = "Invoice date is required") LocalDate date,
        @NotEmpty(message = "Invoice must contain at least one line") @Valid List<InvoiceLine> lines) {
}

File: frontend/src/types/invoice.ts
export interface InvoiceLine {
  description: string;
  currency: string;
  amount: number;
}

export interface Invoice {
  currency: string;
  date: string;
  lines: InvoiceLine[];
}

export interface InvoiceRequest {
  invoice: Invoice;
}

export interface InvoiceLineFormState {
  id: string;
  description: string;
  currency: string;
  amount: string;
}

export interface InvoiceFormState {
  currency: string;
  date: string;
  lines: InvoiceLineFormState[];
}
